File: epic-timeline-backend/src/main/java/com/epicstuff/model/enums/CharacterType.java
package com.epicstuff.model.enums;

public enum CharacterType {
    MORTAL("Mortal"),
    GOD("God"),
    MONSTER("Monster");

    private final String displayName;

    CharacterType(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
    
}

File: epic-timeline-frontend/src/types/CharacterType.ts
export enum CharacterType {
    MORTAL = 'MORTAL',
    GOD = 'GOD',
    MONSTER = 'MONSTER'
}

export const CharacterTypeDisplay: Record<CharacterType, string> = {
    [CharacterType.MORTAL]: 'Mortal',
    [CharacterType.GOD]: 'God',
    [CharacterType.MONSTER]: 'Monster'
};

export type CharacterTypeKey = keyof typeof CharacterType;